writestats.c fails with intel/2023.0.0 icx compiler

The following error was encountered when compiling writestats.c, 
adding header file <ctype.h> as suggested solves the problem.   

```
/glade/work/jedwards/sandboxes/cesm2_x_alpha/components/cism/source_cism/libglimmer/writestats.c:70:10: error: call to undeclared library function 'isalnum' with type 'int (int)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  while (isalnum(resfname[i]) && i < CFG_LEN) i++;
         ^
/glade/work/jedwards/sandboxes/cesm2_x_alpha/components/cism/source_cism/libglimmer/writestats.c:70:10: note: include the header <ctype.h> or explicitly provide a declaration for 'isalnum'
1 error generated.

```

--- a/libglimmer/writestats.c
+++ b/libglimmer/writestats.c
@@ -40,7 +40,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <fcntl.h>
-
+#include <ctype.h>
 
 
 #define CFG_LEN 35

